aiida/cmdline/commands/cmd_database.py

```diff
@@ -8,95 +8,61 @@
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
 """`verdi database` commands."""
+# pylint: disable=unused-argument
 
 import click
 
 from aiida.backends.general.migrations.duplicate_uuids import TABLES_UUID_DEDUPLICATION
 from aiida.cmdline.commands.cmd_verdi import verdi
 from aiida.cmdline.params import options
 from aiida.cmdline.utils import decorators, echo
-from aiida.common import exceptions
 
 
 @verdi.group('database')
 def verdi_database():
-    """Inspect and manage the database."""
+    """Inspect and manage the database.
+
+    .. deprecated:: v3.0.0
+    """
 
 
 @verdi_database.command('version')
 def database_version():
     """Show the version of the database.
 
     The database version is defined by the tuple of the schema generation and schema revision.
-    """
-    from aiida.manage.manager import get_manager
 
-    manager = get_manager()
-    manager._load_backend(schema_check=False)  # pylint: disable=protected-access
-    backend_manager = manager.get_backend_manager()
-
-    echo.echo('Generation: ', bold=True, nl=False)
-    echo.echo(backend_manager.get_schema_generation_database())
-    echo.echo('Revision:   ', bold=True, nl=False)
-    echo.echo(backend_manager.get_schema_version_database())
+    .. deprecated:: v2.1.0
+    """
+    echo.echo_warning(
```

Comment:
We have a dedicated decorator for this. Probably best to use that. Its in `aiida.cmdline.utils.decorators`.

Reply:
Ok, will do. A couple of questions though:

1.  I see it takes care of the warning, but should I still leave the `.. deprecated:: v2.1.0` in the docstring?
2. Do I just decorate the end commands? Just the group? Both the group and the end commands?
3. I didn't find anything on the docs or the wiki for deprecations of the CLI, should I maybe add this info to the [wiki](https://github.com/aiidateam/aiida-core/wiki/Deprecations-and-API-changes)?

Reply:
1. Yes, I think leaving the explicit deprecated in the docstring is ok. But I think it should mention the version [when it was deprecated](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated), not when it is going to be removed.
2. definitely the leaf commands as the group is not really called (just for the help) but if it works, why not also add it there.
3. Feel free to add it to the wiki. Since this is a developer's aspect, it should go there and not the docs.

Reply:
1. I see. But then how do we keep easy track of when we agreed to remove something? I included this info in the deprecation warning now but on second thought I am not sure this is something we want to show to the user necessarily... 🤔 
2. Nah, in the end it did nothing for the groups, not even showed up if I tried using the help. But I think if this is a limitation of the decorator we could maybe extend it to also work on groups no? If it is even possible. For the future anyways.
3. Check out if you think it is clear: [wiki section](https://github.com/aiidateam/aiida-core/wiki/Deprecations-and-API-changes)

Reply:
I think it is fine to mention to the user when we are planning on removing it. Why wouldn't we? This shouldn't be a secret should it and it affects them.